spec/api_auth_spec.rb

```diff
@@ -94,6 +112,32 @@ def hmac(secret_key, request)
       request['date'] = "٢٠١٤-٠٩-٠٨ ١٦:٣١:١٤ +٠٣٠٠"
       expect(ApiAuth.authentic?(request, "123")).to eq false
     end
+
+    context "canonical string contains the http_method" do
+      let(:request){
+        new_request = Net::HTTP::Put.new("/resource.xml?foo=bar&bar=foo",
+          'content-type' => 'text/plain',
+          'content-md5' => '1B2M2Y8AsgTpgAmY7PhCfg==',
+          'date' => Time.now.utc.httpdate
+        )
+        canonical_string = ApiAuth::Headers.new(new_request).canonical_string_with_http_method
+        signature = hmac("123", new_request, canonical_string)
+        new_request["Authorization"] = "APIAuth 1044:#{signature}"
+        new_request
+      }
+
+      it "validates for canonical_strings containing the http_method" do
+        expect(ApiAuth.authentic?(request, "123")).to eq true
+      end
+
+      it "fails to validate if the request method differs" do
+        canonical_string = ApiAuth::Headers.new(request).canonical_string_with_http_method
```

Comment:
Pass in a different method how? Build a whole new request object like Net::HTTP::Post with all the same params that was used for the PUT in the let block?